x/cron/keeper/msg_server_test.go

```diff
@@ -6,12 +6,171 @@ import (
+
+	tests := []struct {
+		name        string
+		msg         types.MsgAddSchedule
+		expectedErr string
+	}{
+		{
+			"empty authority",
+			types.MsgAddSchedule{
+				Authority: "",
+				Name:      "name",
+				Period:    3,
+				Msgs: []types.MsgExecuteContract{
+					{
+						Contract: "contract",
+						Msg:      "msg",
+					},
+				},
+				ExecutionStage: types.ExecutionStage_EXECUTION_STAGE_BEGIN_BLOCKER,
+			},
+			"authority is invalid",
+		},
+		{
+			"invalid authority",
+			types.MsgAddSchedule{
+				Authority: "invalid authority",
+				Name:      "name",
+				Period:    3,
+				Msgs: []types.MsgExecuteContract{
+					{
+						Contract: "contract",
+						Msg:      "msg",
+					},
+				},
+				ExecutionStage: types.ExecutionStage_EXECUTION_STAGE_BEGIN_BLOCKER,
+			},
+			"authority is invalid",
+		},
+		{
+			"invalid name",
+			types.MsgAddSchedule{
+				Authority: testutil.TestOwnerAddress,
+				Name:      "",
+				Period:    3,
+				Msgs: []types.MsgExecuteContract{
+					{
+						Contract: "contract",
+						Msg:      "msg",
+					},
+				},
+				ExecutionStage: types.ExecutionStage_EXECUTION_STAGE_BEGIN_BLOCKER,
+			},
+			"name is invalid",
+		},
+		{
+			"invalid period",
+			types.MsgAddSchedule{
+				Authority: testutil.TestOwnerAddress,
+				Name:      "name",
+				Period:    0,
+				Msgs: []types.MsgExecuteContract{
+					{
+						Contract: "contract",
+						Msg:      "msg",
+					},
+				},
+				ExecutionStage: types.ExecutionStage_EXECUTION_STAGE_BEGIN_BLOCKER,
+			},
+			"period is invalid",
+		},
+		{
+			"empty msgs",
+			types.MsgAddSchedule{
+				Authority:      testutil.TestOwnerAddress,
+				Name:           "name",
+				Period:         3,
+				Msgs:           []types.MsgExecuteContract{},
+				ExecutionStage: types.ExecutionStage_EXECUTION_STAGE_BEGIN_BLOCKER,
+			},
+			"msgs should not be empty",
+		},
+		{
+			"invalid execution stage",
+			types.MsgAddSchedule{
+				Authority: testutil.TestOwnerAddress,
+				Name:      "name",
+				Period:    3,
+				Msgs: []types.MsgExecuteContract{
+					{
+						Contract: "contract",
+						Msg:      "msg",
+					},
+				},
+				ExecutionStage: 7,
+			},
+			"execution stage is invalid",
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			resp, err := msgServer.AddSchedule(ctx, &tt.msg)
+			require.ErrorContains(t, err, tt.expectedErr)
+			require.Nil(t, resp)
+		})
+	}
+}
+
+func TestMsgRemoveScheduleValidate(t *testing.T) {
+	k, ctx := testkeeper.CronKeeper(t, nil, nil)
+	msgServer := cronkeeper.NewMsgServerImpl(*k)
+
+	tests := []struct {
+		name        string
+		msg         types.MsgRemoveSchedule
+		expectedErr string
+	}{
+		{
+			"empty authority",
+			types.MsgRemoveSchedule{
+				Authority: "",
+				Name:      "name",
+			},
+			"authority is invalid",
+		},
+		{
+			"invalid authority",
+			types.MsgRemoveSchedule{
+				Authority: "",
+				Name:      "name",
+			},
+			"authority is invalid",
+		},
```

Comment:
this test case is just the same as the previous one. I think you mean to provide
```go
Authority: "invalid authority",
```
here just like you did in the `TestMsgAddScheduleValidate`?

Reply:
Correct! Thx!